cmd/argocd-server/commands/argocd_server.go

```diff
@@ -123,6 +125,20 @@ func NewCommand() *cobra.Command {
 				baseHRef = rootPath
 			}
 
+			tracerProvider := trace.NewNoopTracerProvider()
+			if otlpAddress != "" {
+				var closer func()
+				var err error
+				tracerProvider, closer, err = cmdutil.InitTracer("argocd-server", otlpAddress)
+				if err != nil {
+					log.Fatalf("failed to initialize tracing: %v", err)
+				}
+				defer closer()
+			}
+			// TODO(yeya24): now we register the tracerProvider globally so it can be used by grpc tracing.
+			// Pass it to ArgoCD server for more fine grained tracing.
+			_ = tracerProvider
```

Comment:
I would prefer removing this line as it can introduce confusion. By the time we want to implement finer grained tracing we will just make usage of `tracerProvider`.

Reply:
Removed thanks.